### PR TITLE
add collections panel shortcut

### PR DESCRIPTION
**Pull request by jamesgorrie**
Seems logical.

### CODE REVIEW

**Comment by akash1810** [comment]
what's the behaviour when localStorage is empty?

File: kahuna/public/js/search/index.js [hunk 1]
```diff
@@ -109,6 +115,7 @@ search.config(['$stateProvider', '$urlMatcherFactoryProvider',
                 // keep the shortcut keys here to stop overriding
                 return new Map([
                     ['metadataPanel', 'm'],
+                    ['collectionsPanel', 'c'],
```
Comment (theefer) [inline kahuna/public/js/search/index.js]: Only concern is C is "crop" on the image page, and recent talks were hinting at possibly bringing that elsewhere, on the results page perhaps. Would be nice to have a letter that we could reuse elsewhere even on the image page for instance.

Comment (theefer) [inline kahuna/public/js/search/index.js]: L?

Comment (jamesgorrie) [inline kahuna/public/js/search/index.js]: B for buckets?

**Comment by jamesgorrie** [comment]
@akash1810 not sure this touches localStorage?

File: kahuna/public/js/search/index.js (hunk 1 above)
Comment (theefer) [inline kahuna/public/js/search/index.js]: Can we put bucket (and basket and tag and route) behind us and not bring them back please? :pray:

Comment (jamesgorrie) [inline kahuna/public/js/search/index.js]: :trollface:

**Comment by akash1810** [comment]
> not sure this touches localStorage?

Sorry, I meant more what's the behaviour when the `showCollectionsPanel` is not set in localStorage - i.e. would the shortcut work if the collections panel hasn't been enabled?

**Comment by jamesgorrie** [comment]
@akash1810 yip, that's all dealt with inside the panel controller.

File: kahuna/public/js/search/index.js (hunk 1 above)
Comment (jamesgorrie) [inline kahuna/public/js/search/index.js]: I like `l` -> changed.

**Comment by jamesgorrie** [comment]
Anyone?

**Comment by akash1810** [comment]
:+1:

**Comment by theefer** [comment]
:+1:

**Comment by paperboyo** [comment]
:-( I thought panel shortcuts for showing/hiding/locking should be independent of the features housed in these panels and be consistent among tools. It would still be possible to have shortcuts to specific features (would take one to the panel scrolled to that feature in case there are more features housed in the same panel).

That stems from the model of panels having features and not every feature having/being its own panel.

**Comment by jamesgorrie** [comment]
> not every feature having/being its own panel.

This isn't something we have at the moment.
If the collections moves into a sub-panel (or something of the like) - it's still nice to have a shortcut no?

**Comment by paperboyo** [comment]
Keyboard shortcuts = always nice :clap:

Keyboard shortcuts to specific features = very good!.

Occupying keyboard with specific shortcuts before deciding on the general ones = dangerous, prone to conflicts/reversals, no?

But, I may have (wrongly?) assumed that `l` - coming from co`l`lections - will gonna stay as a shortcut to the whole panel(*). If not - cool!

(*)I hope it doesn't come from `l`eft, because then I will have to rip out my `r` key and stick where `;` is now :laughing:

**Comment by jamesgorrie** [comment]
> But, I may have (wrongly?) assumed that l - coming from collections - will gonna stay as a shortcut to the whole panel(*). If not - cool!

I wouldn't say so - it'll be linked to collections.

I was assuming, because `c`rop existed, this was co`l`lections.

**Comment by paperboyo** [comment]
Sweet! :candy:

**Comment by theefer** [comment]
@paperboyo may have a point on keyboard shortcut to feature vs panel. This would prevent us from making `l` a shortcut to *add to a collection* (current image, selection, whatnot), which seems a much more useful case for a shortcut than show/hide a panel. Something to think about before it gets used too much.

**Comment by paperboyo** [comment]
> Something to think about before it gets used too much.

My main (only?) point!

A good model, I think, is to have them logical, e.g.:
Feature `X` > Modified feature `Shift + X` > Feature action `Ctrl + X` > Modified feature action `Ctrl + Alt + X`> Even more `Ctrl + Alt + Shift + X`. Not that I have _clear_ examples.

But before that should come general cross-tools UI convention shortcuts, I think.

Usage of Ctrl + Alt should be mindful of https://github.com/guardian/scribe/issues/448 on Windows, though...

**Pull request merged**